File: testdata/bench/go/BenchmarkTest00085.go
package bench

import (
	"net/http"
	"text/template"
)

// Template injection via Header
func Handler00085(w http.ResponseWriter, r *http.Request) {
	tmplStr := r.Header.Get("X-Template")
	t, _ := template.New("t").Parse(tmplStr)
	t.Execute(w, nil)
}

File: testdata/bench/kotlin/BenchmarkTest00085.kt
package bench

import io.ktor.server.application.*
import io.ktor.server.request.*
import io.ktor.server.response.*
import java.io.ObjectInputStream
import java.net.URL

// Vulnerable: ObjectInputStream from URL provided by user
suspend fun handler00085(call: ApplicationCall) {
    val url = call.parameters["url"]
    val stream = URL(url!!).openStream()
    val ois = ObjectInputStream(stream)
    val obj = ois.readObject()
    call.respondText(obj.toString())
}
